src/main/scala/eu/proteus/job/operations/data/serializer/SAXResultKryoSerializer.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 The Proteus Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.proteus.job.operations.data.serializer
+
+import com.esotericsoftware.kryo.Kryo
+import com.esotericsoftware.kryo.Serializer
+import com.esotericsoftware.kryo.io.Input
+import com.esotericsoftware.kryo.io.Output
+import eu.proteus.job.operations.data.results.SAXResult
+
+/**
+ * Kryo serializer for SAX results.
+ * {{{class SAXResult(coilId: Int, x1: Long, x2: Long, classId: String, similarity: Double)}}}
+ */
+class SAXResultKryoSerializer
+  extends Serializer[SAXResult]
+  with Serializable {
+
+  override def write(kryo: Kryo, output: Output, r: SAXResult): Unit = {
```

Comment:
Could we optimize data types in the serialization process?

I suggest the followings ones:
* **coilId**: integer
* **varId**: integer (1-56). Note that you should parse the original string value (CXXXX) to an integer.
* **x1**: double (you are currently using long, and some original x values are double).
* **x2**: double (same as above)
* **classId**: byte
* **similarity**: double